File: backend/src/main/java/io/makewebsite/entity/Message.java
package io.makewebsite.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.GenericGenerator;

import java.time.LocalDateTime;
import java.util.UUID;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
@Table(name = "messages")
public class Message {

    @Id
    @GeneratedValue(generator = "UUID")
    @GenericGenerator(name = "UUID", strategy = "org.hibernate.id.UUIDGenerator")
    @Column(updatable = false)
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "boutique_id", nullable = false)
    private Boutique boutique;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "conversation_id")
    private Conversation conversation;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "customer_id")
    private Customer customer;

    @Size(max = 150)
    @Column(name = "customer_name")
    private String customerName;

    @Size(max = 150)
    @Column(name = "customer_email")
    private String customerEmail;

    @Size(max = 20)
    @Column(name = "customer_phone")
    private String customerPhone;

    @NotNull
    @Size(max = 10)
    @Column(name = "sender_role", nullable = false)
    private String senderType;

    @NotNull
    @Column(name = "content", nullable = false, columnDefinition = "TEXT")
    private String content;

    @Builder.Default
    @Column(name = "is_read")
    private Boolean isRead = false;

    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
    }
}

File: frontend/lib/models/message.dart
class Message {
  final String id;
  final String conversationId;
  final String senderType;
  final String content;
  final bool isRead;
  final String? createdAt;

  Message({
    this.id = '',
    required this.conversationId,
    required this.senderType,
    required this.content,
    this.isRead = false,
    this.createdAt,
  });

  factory Message.fromJson(Map<String, dynamic> json) => Message(
    id: json['messageId']?.toString() ?? json['id']?.toString() ?? '',
    conversationId: (json['conversationId'] ?? '').toString(),
    senderType: json['senderType'] ?? 'CUSTOMER',
    content: json['content'] ?? '',
    isRead: json['isRead'] ?? false,
    createdAt: json['createdAt'],
  );
}
